File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n);
        int pos = n-1;
        int left = 0;
        int right= n-1;
        while(left<=right){
            int left_sq = nums[left]*nums[left];
            int right_sq = nums[right]* nums[right];
            if(left_sq >= right_sq){
                result[pos] = left_sq;
                left++;
               
            }
            else{
                result[pos] = right_sq;
                right--;
            }
            pos--;
        }
        
        return result;
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l = []
        for i in range(n):
            l.append(0)

        left = 0
        right = n-1
        pos=n-1
        while(left<=right):
            left_sq = nums[left]*nums[left]
            right_sq= nums[right]*nums[right]
            if(left_sq >= right_sq):
                l[pos] = left_sq
                left += 1
            
            else:
                l[pos] = right_sq
                right -= 1

            pos -= 1
        return l
        